### PR TITLE
VS Code: add characters logger stats to `fixup.apply:succeeded` events

### PR DESCRIPTION
**Pull request by valerybugakov**
- Enhances `fixup.apply:succeeded` events by adding boolean fields to improve PCW tracking and avoid double-counting Cody's contributions. The updated logic now leverages the `CharactersLogger` logic for classifying document changes. 
    - For example, when the Edit command produces a change outside of visible ranges, it is flagged by both the Edit command logic and `CharactersLogger` telemetry. This ensures that non-visible changes, previously excluded due to non-manual nature, can now be accurately counted in PCW calculations because they are flagged accordingly.
    - The updated event structure can be found here: `vscode/test/e2e/__snapshots__/command-edit.test.ts/edit-fixup-task-1.txt`
- @kelsey-brown, I made some changes to the existing `cody.characters:flush` event in this PR:
    - Split the `outside_of_visible_ranges` category into two new types: `partially_outside_of_visible_ranges` and `fully_outside_of_visible_ranges`.
    - Added a new `disjoint_change` category that we can filter out from manually written code. This happens when a document change event includes edits in multiple separate parts of the document. It can be produced with find-and-replace operations, but not when someone's typing code by hand. Also, the Edit command often produces this type of document change event.
- Part of https://linear.app/sourcegraph/issue/CODY-4197/add-metadata-to-cody-code-gen-analytics-events-to-prevent-double. See more context and related discussion there.
- Closes https://linear.app/sourcegraph/issue/CODY-4199/add-char-counts-metadata-to-fixup-events

## Test plan

CI + updated unit tests + new assertions in the Edit command E2E tests.

### CODE REVIEW

File: vscode/test/e2e/command-edit.test.ts-snapshots/edit-fixup-task-1-darwin.txt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        "value": 0
+      },
+      {
+        "key": "charsInserted",
+        "value": 46
+      },
+      {
+        "key": "charsDeleted",
+        "value": 90
+      },
```
Comment (valerybugakov) [inline vscode/test/e2e/command-edit.test.ts-snapshots/edit-fixup-task-1-darwin.txt:78]: New fields added to `cody.fixup.apply:succeeded` events.

Comment (valerybugakov) [reply vscode/test/e2e/command-edit.test.ts-snapshots/edit-fixup-task-1-darwin.txt]: UPD: removed `isUndo` and `isRedo` from this list.

File: vscode/test/e2e/__snapshots__/command-edit.test.ts/edit-fixup-task-1.txt [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        "value": 0
+      },
+      {
+        "key": "charsInserted",
+        "value": 46
+      },
+      {
+        "key": "charsDeleted",
+        "value": 90
+      },
```
Comment (valerybugakov) [inline vscode/test/e2e/__snapshots__/command-edit.test.ts/edit-fixup-task-1.txt:70]: New fields added to `cody.fixup.apply:succeeded` events.

**Comment by kelsey-brown** [comment]
LGTM from an analytics POV. For

> Split the outside_of_visible_ranges category into two new types: partially_outside_of_visible_ranges and fully_outside_of_visible_ranges.

I should still exclude both of these from the "all inserted characters" metric by default, since they are not "user-inserted", and then add them back if they are generated by cody, right? 

I'll update the code for the metric on my end once this is deployed.

**Review by kelsey-brown — APPROVED** [review]

**Comment by valerybugakov** [comment]
> I should still exclude both of these from the "all inserted characters" metric by default, since they are not "user-inserted", and then add them back if they are generated by cody, right?

Yes, I added these groups to help with debugging issues.

**Comment by valerybugakov** [comment]
I'm going to merge this one. Will be happy to address any potential comments in a follow-up PR.

**Pull request merged**